### ISSUE TITLE
Investigate why global event handlers are broken on 2.3 and master

### ISSUE THREAD

**Issue by mvdan** [TykTechnologies/tyk#1269]
And fix them, adding tests.

**Comment by matiasinsaurralde** [comment]
Interesting, it will be useful for #1260 as I plan to adapt those changes to support event handlers.

**Comment by mvdan** [comment]
@matiasinsaurralde is already working on stuff very close to this (and the handlers seem to work for him), and I hadn't started this, so passing it on.

**Comment by buger** [comment]
Moving to 1.4.1

**Comment by matiasinsaurralde** [comment]
Yes, it’s on my list today

**Comment by buger** [comment]
@matiasinsaurralde bump :)

**Comment by matiasinsaurralde** [comment]
I've been investigating this issue again. I see that we have a `config.Global.EventHandlers` field, this is loaded from `tyk.conf` and has the same type as the event handlers block we use for API definitions (`apidef.EventHandlerMetaConfig`):
```go
	EventHandlers                     apidef.EventHandlerMetaConfig         `json:"event_handlers"`
	EventTriggers                     map[apidef.TykEvent][]TykEventHandler `json:"event_trigers_defunct"`
	PIDFileLocation                   string                                `json:"pid_file_location"`
```
However I don't see any references to `config.Global.EventHandlers`, this means we never load these "global" events.
In the other side, event handlers seem to be working fine when attached to API definitions.
I don't see any reference to "global" event handlers in our  docs, was this feature available in the past? I'm investigating previous versions at the moment to see if we broke it somehow, perhaps after configuration tweaks?

**Comment by buger** [comment]
@matiasinsaurralde indeed weird, try to check 2.4 branch on GitHub. 

If it is not in 2.4, estimate how hard is to add them, and maybe it just make sense remove them from all the configs and docs.

**Comment by matiasinsaurralde** [comment]
I've realized that the original issue was about JSVM event handlers, I can confirm that they stopped working after 2.3.
To reproduce:
- Modify `tyk.conf` with `"enable_jsvm": true`
- Setup a protected API
- Add event handlers to it:
```json
"event_handlers": {
    "events": {
        "AuthFailure": [
                {
                    "handler_name":"eh_dynamic_handler",
                    "handler_meta": {
                        "name": "sampleHandler",
                        "path": "event_handlers/sample_handler.js"
                    }
                }
            ]
    }
}
```
- Create `event_handlers/sample_handler.js` with the following contents:
```js
var sampleHandler = new TykJS.TykEventHandlers.NewEventHandler({});
log("sampleHandler is loaded")
sampleHandler.NewHandler(function(event, context) {
    log("sampleHandler is called")
    return
});
```
- Start the gateway, you will see the event handler code is loaded correctly:
```
INFO jsvm: Loading JS File: event_handlers/sample/sample_event_handler.js
INFO jsvm-logmsg: sampleHandler is loaded type=log-msg
```
- Send a request to the API triggering a failed authentication:
```
curl http://localhost:8080/quickstart/headers -H 'Authorization: badtoken'
```
- The authentication will trigger an `AuthFailure` event but the JS handler won't be called

I can confirm that `JSVMEventHandler.HandleEvent` is called with the right parameters (the event message):
```
JSVMEventHandler.HandleEvent:  {AuthFailure {{Auth Failure R0VUIC9xdWlja3N0YXJ0Mi9oZWFkZXJzIEhUVFAvMS4xDQpIb3N0OiBsb2NhbGhvc3Q6ODA4MA0KVXNlci1BZ2VudDogY3VybC83LjU2LjENCkFjY2VwdDogKi8qDQpBdXRob3JpemF0aW9uOiAxMjM0NQ0KDQo=} /quickstart/headers ::1 12345} 2018-05-10 16:24:18.258555696 -0400 -04}
```
Then `sampleHandler.DoProcessEvent` is called and the JSVM reports that it's not defined.

**Comment by matiasinsaurralde** [comment]
#1693 should fix this.

**Comment by buger** [comment]
Released as part of 2.7
